src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1346,36 +1355,58 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 		return this.workspaceEditingService.updateFolders(0, workspaceCreationData.length, workspaceCreationData);
 	}
 
-	private async doHandleExplorerDrop(source: ExplorerItem, target: ExplorerItem, isCopy: boolean): Promise<void> {
+	private async doHandleExplorerDrop(sources: ExplorerItem[], target: ExplorerItem, isCopy: boolean): Promise<void> {
 		// Reuse duplicate action if user copies
+		const sourceTargetPairs: { source: URI; target: URI }[] = [];
 		if (isCopy) {
 			const incrementalNaming = this.configurationService.getValue<IFilesConfiguration>().explorer.incrementalNaming;
-			const stat = await this.workingCopyFileService.copy(source.resource, findValidPasteFileTarget(this.explorerService, target, { resource: source.resource, isDirectory: source.isDirectory, allowOverwrite: false }, incrementalNaming));
-			if (!stat.isDirectory) {
-				await this.editorService.openEditor({ resource: stat.resource, options: { pinned: true } });
+			for (const source of sources) {
+				sourceTargetPairs.push({ source: source.resource, target: findValidPasteFileTarget(this.explorerService, target, { resource: source.resource, isDirectory: source.isDirectory, allowOverwrite: false }, incrementalNaming) });
 			}
+			const stats = await this.workingCopyFileService.copy(sourceTargetPairs);
+			stats.forEach(async stat => {
+				if (!stat.isDirectory) {
+					await this.editorService.openEditor({ resource: stat.resource, options: { pinned: true } });
+				}
+			});
 
 			return;
 		}
 
 		// Otherwise move
-		const targetResource = joinPath(target.resource, source.name);
-		if (source.isReadonly) {
+		for (const source of sources) {
 			// Do not allow moving readonly items
-			return Promise.resolve();
+			if (!source.isReadonly) {
+				sourceTargetPairs.push({ source: source.resource, target: joinPath(target.resource, source.name) });
+			}
 		}
 
 		try {
-			await this.workingCopyFileService.move(source.resource, targetResource);
+			await this.workingCopyFileService.move(sourceTargetPairs);
 		} catch (error) {
 			// Conflict
 			if ((<FileOperationError>error).fileOperationResult === FileOperationResult.FILE_MOVE_CONFLICT) {
-				const confirm = getFileOverwriteConfirm(source.name);
+
+				let confirm: IConfirmation;
+				const overwrites: URI[] = [];
+				for (const { target } of sourceTargetPairs) {
+					const exist = await this.fileService.exists(target);
+					if (exist) {
+						overwrites.push(target);
+					}
+				}
+
+				if (overwrites.length > 1) {
```

Comment:
Maybe the helper method gets all the overwrites and then just creates an appropriate confirm.
So you do not need the length > 1 logic outside the method